test/lib/ansible_test/_internal/core_ci.py

```diff
@@ -306,20 +306,24 @@ def get(self, tries=3, sleep=15, always_raise_on=None):
             )
         else:
             response_json = response.json()
-
             status = response_json['status']
-            con = response_json['connection']
-
-            self.connection = InstanceConnection(
-                running=status == 'running',
-                hostname=con['hostname'],
-                port=int(con.get('port', self.port)),
-                username=con['username'],
-                password=con.get('password'),
-            )
-
-            if self.connection.password:
-                display.sensitive.add(str(self.connection.password))
+            con = response_json.get('connection')
+
+            if con:
+                self.connection = InstanceConnection(
+                    running=status == 'running',
+                    hostname=con['hostname'],
+                    port=int(con['port']) if con.get('port') else None,
```

Comment:
Preserve the original behavior for the default port. Add `self.port = None` under the `elif self.provider == 'vmware':` conditional in `__init__` above to provide the correct default.

```suggestion
                    port=int(con.get('port', self.port)),
```